Source/GameView/Repositories/GameAPIRepo.swift

```diff
@@ -78,58 +93,83 @@ public class GameAPIRepo: GameRepository {
     
   }
   
-  //TODO: Replace gameList with individual game and make the model call the function many times.
   public func getGameDetails(of gameList: [Game], with serviceCaller: ServiceCaller) {
     
-    let dispatchGroup = DispatchGroup()
+    let operationQueue = OperationQueue()
     
-    gameList.forEach({ (game) in
-      
-      dispatchGroup.enter()
+    let serviceCallOperation = createServiceCallBlockOperation(using: serviceCaller,
+                                                               forGamesList: gameList)
+    
+    let notifyOperation = BlockOperation {
+      self.notifyGameDetailsLoaded(withData: gameList)
+    }
+    
+    notifyOperation.addDependency(serviceCallOperation)
+    
+    operationQueue.addOperation(serviceCallOperation)
+    operationQueue.addOperation(notifyOperation)
+    
+  }
+  
+  private func createServiceCallBlockOperation(using serviceCaller: ServiceCaller,
+                                               forGamesList gameList: [Game]) -> BlockOperation {
+    
+    let serviceCallOperation = BlockOperation {
       
-      let dataBundle = DataBundle()
-      dataBundle.extraData["game"] = game
+      let dispatchGroup = DispatchGroup()
       
-      // MARK: - Call Succeeded Closure
-      serviceCaller.callSucceeded = { [weak self] data, dataBundle in
+      gameList.forEach({ (game) in
+        
+        dispatchGroup.enter()
         
-        guard let game = dataBundle.extraData["game"] as? Game else {
-          return
+        let dataBundle = DataBundle()
+        dataBundle.extraData[gameDataBundleKeys.game] = game
+        
+        // MARK: - Call Succeeded Closure
+        serviceCaller.callSucceeded = { [weak self] data, dataBundle in
+          
+          guard let self = self else {
+            dispatchGroup.leave()
+            return
+          }
+          
+          guard let game = dataBundle.extraData[gameDataBundleKeys.game] as? Game else {
+            dispatchGroup.leave()
+            return
+          }
+          
+          self.decodeGameDetails(using: data, basedOn: game)
+          
+          dispatchGroup.leave()
+          
         }
         
-        guard let self = self else {
-          return
+        // MARK: - ---------------------------
+
+        // MARK: - Call Failed Closure
+        serviceCaller.callFailed = { error in
+          dispatchGroup.leave()
         }
         
-       self.decodeGameDetails(using: data, basedOn: game)
+        // MARK: - ---------------------------
         
-      }
-      
-      // MARK: - ---------------------------
-
-      // MARK: - Call Failed Closure
-      serviceCaller.callFailed = { error in
+        let urlString = "\(SteamURLComponents.specificGameURL)\(game.appID)"
+        let url: URL = URL(string: urlString)!
```

Comment:
Force unwrapping here, remove if possible

Reply:
I could unwrap it, however my thinking is that I know there is something there, and even if it can't access SteamURLComponents well...there will be an error thrown in the form of a ServiceCall.MalformedRequest